gtsam/slam/tests/testEssentialMatrixWithCalibrationFactor.cpp

```diff
@@ -0,0 +1,455 @@
+/**
+ * @file testEssentialMatrixWithCalibrationFactor.cpp
+ * @brief Test EssentialMatrixWithCalibrationFactor class
+ * @author Ayush Baid
+ * @author Akshay Krishnan
+ * @date April 22, 2021
+ */
+
+#include <CppUnitLite/TestHarness.h>
+#include <gtsam/base/Testable.h>
+#include <gtsam/base/numericalDerivative.h>
+#include <gtsam/geometry/Cal3Bundler.h>
+#include <gtsam/geometry/Cal3_S2.h>
+#include <gtsam/geometry/CalibratedCamera.h>
+#include <gtsam/nonlinear/ExpressionFactor.h>
+#include <gtsam/nonlinear/LevenbergMarquardtOptimizer.h>
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
+#include <gtsam/nonlinear/expressionTesting.h>
+#include <gtsam/slam/EssentialMatrixWithCalibrationFactor.h>
+#include <gtsam/slam/dataset.h>
+
+using namespace std;
+using namespace gtsam;
+
+// Noise model for first type of factor is evaluating algebraic error
+noiseModel::Isotropic::shared_ptr model1 =
+    noiseModel::Isotropic::Sigma(1, 0.01);
+// Noise model for second type of factor is evaluating pixel coordinates
+noiseModel::Unit::shared_ptr model2 = noiseModel::Unit::Create(2);
+
+// The rotation between body and camera is:
+gtsam::Point3 bX(1, 0, 0), bY(0, 1, 0), bZ(0, 0, 1);
+gtsam::Rot3 cRb = gtsam::Rot3(bX, bZ, -bY).inverse();
+
+namespace example1 {
+
+const string filename = findExampleDataFile("5pointExample1.txt");
+SfmData data;
+bool readOK = readBAL(filename, data);
+Rot3 c1Rc2 = data.cameras[1].pose().rotation();
+Point3 c1Tc2 = data.cameras[1].pose().translation();
+// TODO: maybe default value not good; assert with 0th
+Cal3_S2 trueK = Cal3_S2();
+// PinholeCamera<Cal3_S2> camera2(data.cameras[1].pose(), trueK);
+Rot3 trueRotation(c1Rc2);
+Unit3 trueDirection(c1Tc2);
+EssentialMatrix trueE(trueRotation, trueDirection);
+
+double baseline = 0.1;  // actual baseline of the camera
+
+Point2 pA(size_t i) { return data.tracks[i].measurements[0].second; }
```

Comment:
Maybe use a good name for this? pA looks like a variable more than a function. gePixelFromA or something? Also consider passing in data instead of using global variables. Same for other variables below.

Reply:
Please keep naming and mve back to old file, where this terminology is being used.